src/SettingsSystem.cpp

```diff
@@ -96,6 +149,70 @@ bool SettingsSystem::reload_audio_settings() {
     return success;
 }
 
+bool SettingsSystem::reload_physics_settings() {
+    bool success = true;
+    std::vector<float> gravity;
+    success &= load_key<std::vector<float>>(KEYS_GRAVITY, gravity);
+
+    if (success) {
+        if (gravity.size() != 3) {
+            std::cerr << "Could not parse sequence of size " << gravity.size() << "as PxVec3" << std::endl;
+            return false;
+        }
+
+        physics_settings_.gravity = {gravity.at(0), gravity.at(1), gravity.at(2)};
+    }
+
+    // update smoothing data
+    float accel_rise = physics_settings_.g_pad_smoothing_data.mRiseRates[0];
+    float brake_rise = physics_settings_.g_pad_smoothing_data.mRiseRates[1];
+    float handbrake_rise = physics_settings_.g_pad_smoothing_data.mRiseRates[2];
+    float steer_left_rise = physics_settings_.g_pad_smoothing_data.mRiseRates[3];
+    float steer_right_rise = physics_settings_.g_pad_smoothing_data.mRiseRates[4];
+    float accel_fall = physics_settings_.g_pad_smoothing_data.mFallRates[0];
+    float brake_fall = physics_settings_.g_pad_smoothing_data.mFallRates[1];
+    float handbrake_fall = physics_settings_.g_pad_smoothing_data.mFallRates[2];
+    float steer_left_fall = physics_settings_.g_pad_smoothing_data.mFallRates[3];
+    float steer_right_fall = physics_settings_.g_pad_smoothing_data.mFallRates[4];
+    success &= load_key<float>(KEYS_ACCEL_RISE, accel_rise);
+    success &= load_key<float>(KEYS_BRAKE_RISE, brake_rise);
+    success &= load_key<float>(KEYS_HANDBRAKE_RISE, handbrake_rise);
+    success &= load_key<float>(KEYS_STEER_LEFT_RISE, steer_left_rise);
+    success &= load_key<float>(KEYS_STEER_RIGHT_RISE, steer_right_rise);
+    success &= load_key<float>(KEYS_ACCEL_FALL, accel_fall);
+    success &= load_key<float>(KEYS_BRAKE_FALL, brake_fall);
+    success &= load_key<float>(KEYS_HANDBRAKE_FALL, handbrake_fall);
+    success &= load_key<float>(KEYS_STEER_LEFT_FALL, steer_left_fall);
+    success &= load_key<float>(KEYS_STEER_RIGHT_FALL, steer_right_fall);
+
+    if (success) {
```

Comment:
Instead of putting an `if (success)` and doing an action, the defuat shoudl be doing an action, and `if !success`, throw an error and don't update settings.